cmd/kubeadm/app/cmd/token.go

```diff
@@ -277,22 +291,65 @@ func RunListTokens(out io.Writer, errW io.Writer, client clientset.Interface) er
 		return errors.Wrap(err, "failed to list bootstrap tokens")
 	}
 
-	w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
-	fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
+	var tokens []*kubeadmapi.BootstrapToken
 	for _, secret := range secrets.Items {
-
 		// Get the BootstrapToken struct representation from the Secret object
 		token, err := kubeadmapi.BootstrapTokenFromSecret(&secret)
 		if err != nil {
 			fmt.Fprintf(errW, "%v", err)
 			continue
 		}
+		tokens = append(tokens, token)
+	}
+
+	if outputFormat == "text" {
+		w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
+		fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
+		for _, token := range tokens {
+			// Get the human-friendly string representation for the token
+			humanFriendlyTokenOutput := humanReadableBootstrapToken(token)
+			fmt.Fprintln(w, humanFriendlyTokenOutput)
+		}
+		w.Flush()
+	} else if outputFormat == "yaml" || outputFormat == "json" {
+		clusterConfig, err := cmdutil.GetClusterConfig(kubeConfigFile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to load cluster config from %s", kubeConfigFile)
+		}
+
+		publicKeyPins, err := cmdutil.GetCAPubKeyPins(clusterConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get CA certs hashes from cluster config")
+		}
+
+		tokenList := BootstrapTokenList{
+			PublicKeyPins:        publicKeyPins,
+			ControlPlaneHostPort: strings.Replace(clusterConfig.Server, "https://", "", -1),
+			Tokens:               tokens,
+		}
+
+		items := struct {
+			BootstrapTokenList BootstrapTokenList `json:"bootstrapTokenList" yaml:"bootstrapTokenList"`
+		}{BootstrapTokenList: tokenList}
+		var output string
 
-		// Get the human-friendly string representation for the token
-		humanFriendlyTokenOutput := humanReadableBootstrapToken(token)
-		fmt.Fprintln(w, humanFriendlyTokenOutput)
+		if outputFormat == "yaml" {
+			yml, err := yaml.Marshal(items)
```

Comment:
You can marshal this into JSON first, and then use `JSONToYAML` (from `sigs.k8s.io/yaml`) in order to get rid of the separate `yaml` tags in `BootstrapTokenList`.
I am not concerned with the different ordering of fields between YAML and JSON serialization, simply because these are not meant to be read by a human in the general case.

Reply:
i think @bart0sh was not happy with the ordering difference between yaml and json. he should confirm if this is still a problem if JSONToYAML is used.

Reply:
I tried that in one of the first versions of this PR. JSONToYAML sorts keys in yaml output  alphabetically, which makes it look different than json output. I switchet to yaml.v2 to fix this as @neolit123 suggested.

Reply:
@rosti > I am not concerned with the different ordering of fields between YAML and JSON

I was. However, the structure has changed quite substantial since that, so the output can look less ugly. It makes sense to try suing JSONToYAML again. Thank you for pointing that out to me.

Reply:
done, please review